### PR TITLE
MRG, FIX: Allow multiple runs of Epochs.drop_bad() even if only using a subset of initial reject channels

### LINKED ISSUE #9483: Trying to apply rejection thresholds repeatedly to Epochs crashes if only using a subset of initial reject channels
MWE:
```python
from pathlib import Path
import mne


sample_dir = Path(mne.datasets.sample.data_path())
sample_fname = sample_dir / 'MEG' / 'sample' / 'sample_audvis_raw.fif'

raw = mne.io.read_raw_fif(sample_fname)
raw.crop(tmax=60)

events = mne.find_events(raw, stim_channel='STI 014')
event_id = {'auditory/left': 1, 'auditory/right': 2, 'visual/left': 3,
            'visual/right': 4, 'face': 5, 'buttonpress': 32}

epochs = mne.Epochs(raw, events=events, event_id=event_id,
                    tmin=-0.2, tmax=0.5, baseline=(None, 0),
                    preload=True)

# This works as expected
reject_a = dict(grad=8e-10, mag=10e-11)
epochs.drop_bad(reject=reject_a)

# This raises an exception, but I'd expect it to work:
# ValueError: reject["mag"] == inf > 1e-10 (old value), new
# reject values must be at least as stringent as previous ones
reject_b = dict(grad=1e-10)
epochs.drop_bad(reject=reject_b)

# This works as expected
reject_c = dict(grad=1e-10, mag=9e-11)
epochs.drop_bad(reject=reject_c)
```

### PR DESCRIPTION
**Pull request by hoechenberger**
Closes #9483

### CODE REVIEW

File: doc/changes/latest.inc [hunk 1]
```diff
@@ -85,6 +85,8 @@ Bugs
 
 - :func:`mne.concatenate_raws` now raises an exception if ``raw.info['dev_head_t']`` differs between files. This behavior can be controlled using the new ``on_mismatch`` parameter (:gh:`9438` by `Richard Höchenberger`_)
 
+- :meth:`mne.Epochs.drop_bad` now lets you reject multiple times, even when only specifying thresholds for a subset of the channel types used in a previous rejection run. Previously, this would raise an exception (:gh:`9484` by `Richard Höchenberger`_)
```
Comment (drammock) [inline doc/changes/latest.inc:88]:
```suggestion
- Fixed bug in :meth:`mne.Epochs.drop_bad` where subsequent rejections failed if they only specified thresholds for a subset of the channel types used in a previous rejection (:gh:`9484` by `Richard Höchenberger`_).
```

File: mne/epochs.py [hunk 2]
```diff
@@ -746,13 +746,29 @@ def _reject_setup(self, reject, flat):
...
             old = old_reject.get(key, np.inf)
-            new = reject.get(key, np.inf)
+            if key in old_reject and key not in reject:
+                # The newly-passed `reject` parameter contains fewer channel
+                # types than the previous one. For the omitted channel types,
+                # simply re-use the old thresholds
+                new = old
+                reject[key] = new
+            else:
+                new = reject[key]
```
Comment (drammock) [inline mne/epochs.py:756]: possibly simpler / clearer?

```py
# copy prev. keys that weren't passed this time
for key in set(old_reject) - set(reject):
    reject[key] = old_reject[key]
# make sure more stringent
for key in reject:
    if reject[key] > old_reject[key]:
        raise ValueError(...)
```

**Review by drammock — APPROVED** [review]

File: mne/epochs.py (hunk 2 above)
Comment (hoechenberger) [reply mne/epochs.py:756]: Good idea. I didn't want to touch too much of the existing implementation, but I like your proposal. I will test to see if this works.

Comment (hoechenberger) [reply mne/epochs.py]: I've refactored the code based on your suggestion

File: mne/epochs.py [hunk 3]
```diff
@@ -744,34 +744,26 @@ def _reject_setup(self, reject, flat):
...
-            if new < old:
-                raise ValueError(bad_msg.format(kind='flat', key=key,
-                                                new=new, old=old, op='<'))
+        # copy thresholds for channel types that were used previously, but not
+        # passed this time
+        for key in set(old_reject) - set(reject):
+            reject[key] = old_reject[key]
+        # make sure new thresholds are at least as stringent as the old ones
+        for key in reject:
+            if key in old_reject and reject[key] > old_reject[key]:
```
Comment (drammock) [inline mne/epochs.py:754]:
```suggestion
        for key in old_reject:
            if reject[key] > old_reject[key]:
```

File: mne/epochs.py [hunk 4]
```diff
@@ -744,34 +744,26 @@ def _reject_setup(self, reject, flat):
...
+                raise ValueError(
+                    bad_msg.format(kind='reject', key=key, new=reject[key],
+                                   old=old_reject[key], op='>'))
+
+        # same for flat thresholds
+        for key in set(old_flat) - set(flat):
+            flat[key] = old_flat[key]
+        for key in flat:
+            if key in old_flat and flat[key] < old_flat[key]:
```
Comment (drammock) [inline mne/epochs.py:763]:
```suggestion
        for key in old_flat:
            if flat[key] < old_flat[key]:
```

**Review by drammock — APPROVED** [review]
one more suggested simplification

File: mne/epochs.py (hunk 3 above)
Comment (hoechenberger) [reply mne/epochs.py:754]: This breaks, unfortunately, in situations where `old_reject` is empty (i.e., if no previous reject has been done) or it doesn't contain a threshold for a channel type that is now appearing in `reject` (i.e., no previous reject for this specific channel type has been done).

I think, therefore, it's okay to keep as-is, and will proceed to merging, ok?

Comment (drammock) [reply mne/epochs.py:754]: ah, yes. I didn't think about old_reject being empty. Will merge.

**Comment by drammock** [comment]
thanks @hoechenberger!

**Pull request merged**